### PR TITLE
add js and add components from gcs

### PR DESCRIPTION
**Pull request by wprocessor**

### CODE REVIEW

File: js/google_cse.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+(function ($, Drupal, drupalSettings) {
+  Drupal.behaviors.googleCSECustomSearch = {
+    attach: function (context, settings) {
+      var googleCSEWatermark = function (id) {
+        var f = $(id)[0];
+        if (f) {
```
Comment (jbguerraz) [inline js/google_cse.js]: remove the console.log (the whole if(f) block)

File: js/google_cse.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            if (q.value == '') {
+              q.style.background = '#FFFFFF url(https://www.google.com/cse/intl/' + drupalSettings.googleCSE.language + '/images/google_custom_search_watermark.gif) left no-repeat';
+            }
+          };
+          var f = function () {
+            q.style.background = '#ffffff';
+          };
+          q.onfocus = f;
+          q.onblur = b;
+  //      if (!/[&?]query=[^&]/.test(l.search)) {
```
Comment (jbguerraz) [inline js/google_cse.js]: to be removed

File: js/google_cse.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+          b();
+  //      }
```
Comment (jbguerraz) [inline js/google_cse.js]: to be removed as well )

File: google_cse.module [hunk 4]
```diff
@@ -221,17 +221,32 @@ function google_cse_form_search_form_alter(&$form, &$form_state, $form_id) {
     google_cse_sitesearch_form($form);
     $form['#attributes']['class'][] = 'google-cse';
   }
+
+  $form['#attributes']['class'][] = 'google-cse';
```
Comment (andypost) [inline google_cse.module]: not sure it makes sense, cos added conditionally above

File: google_cse.module [hunk 5]
```diff
@@ -221,17 +221,32 @@ function google_cse_form_search_form_alter(&$form, &$form_state, $form_id) {
...
 }
 
 /**
  * Adds custom submit handler for search block form.
  */
 function google_cse_form_search_block_form_alter(&$form, &$form_state, $form_id) {
+  $config = \Drupal::config('google_cse.settings');
```
Comment (andypost) [inline google_cse.module]: please move inside of `if ()` where it actually used

File: js/google_cse.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        }
+      };
+      googleCSEWatermark('[data-drupal-selector="search-block-form"] [data-drupal-form-fields="edit-keys"]');
+      googleCSEWatermark('.search-block-form.google-cse');
+      googleCSEWatermark('#search-form.google-cse');
+      googleCSEWatermark('#google-cse-results-searchbox-form');
+    }
+  };
+})(jQuery, Drupal, drupalSettings);
```
Comment (andypost) [inline js/google_cse.js]: nit, needs NL

File: src/Form/GoogleCseForm.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $form['keys'] = array(
+      '#type' => 'textfield',
+      '#required' => TRUE,
+    );
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#value' => 'Search',
```
Comment (andypost) [inline src/Form/GoogleCseForm.php]: should be `$this->t('Search')` - translatable

File: src/Form/GoogleCseForm.php [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $config = \Drupal::configFactory()->get('GoogleCSE.settings');
+    $path = '/search/' . $config->get('GoogleCSE.wildcard') . '/' . $form_state->getValues()['keys'];
+    $response = new RedirectResponse($path);
+    $response->send();
```
Comment (andypost) [inline src/Form/GoogleCseForm.php]: no reason to send, just use `$form_state->redirect()`

File: google_cse.module [hunk 9]
```diff
@@ -221,6 +221,9 @@ function google_cse_form_search_form_alter(&$form, &$form_state, $form_id) {
     google_cse_sitesearch_form($form);
     $form['#attributes']['class'][] = 'google-cse';
   }
+
+  google_cse_sitesearch_form($form);
+  $form['#attributes']['class'][] = 'google-cse';
```
Comment (andypost) [inline google_cse.module]: This executed above already and conditionally!!!
suppose this 2 lines should be removed at all

File: google_cse.module [hunk 10]
```diff
@@ -230,8 +233,21 @@ function google_cse_form_search_block_form_alter(&$form, &$form_state, $form_id)
   /** @var \Drupal\search\SearchPageRepositoryInterface $repo */
   $repo = \Drupal::service('search.search_page_repository');
   if ($repo->getDefaultSearchPage() == 'google_cse_search_type') {
+    $config = \Drupal::config('google_cse.settings'); 
     google_cse_sitesearch_form($form);
     $form['#attributes']['class'][] = 'google-cse';
```
Comment (andypost) [inline google_cse.module]: that duplicates code in hunk above, suppose better move this attribute assignment inside `google_cse_sitesearch_form()`

File: google_cse.module [hunk 11]
```diff
@@ -217,10 +218,7 @@ function google_cse_sitesearch_default() {
  * Adds custom submit handler for search form.
  */
 function google_cse_form_search_form_alter(&$form, &$form_state, $form_id) {
-  if ($form['module']['#value'] == 'google_cse') {
```
Comment (andypost) [inline google_cse.module:220]: that's wrong removal!

File: google_cse.module [hunk 12]
```diff
@@ -218,7 +218,9 @@ function google_cse_sitesearch_default() {
  * Adds custom submit handler for search form.
  */
 function google_cse_form_search_form_alter(&$form, &$form_state, $form_id) {
-  google_cse_sitesearch_form($form);
+  if ($form_id == 'search_form') {
```
Comment (andypost) [inline google_cse.module]: Use this http://cgit.drupalcode.org/drupal/tree/core/modules/search/src/Form/SearchBlockForm.php#n77
to check what page is target in the block

File: google_cse.module [hunk 13]
```diff
@@ -217,9 +218,8 @@ function google_cse_sitesearch_default() {
  * Adds custom submit handler for search form.
  */
 function google_cse_form_search_form_alter(&$form, &$form_state, $form_id) {
-  if ($form['module']['#value'] == 'google_cse') {
+  if ($form_id == 'search_form') {
```
Comment (andypost) [inline google_cse.module]: It should be http://cgit.drupalcode.org/drupal/tree/core/modules/search/src/Form/SearchBlockForm.php#n77

```
if ($default_page == 'google_cse')
```

**Comment by andypost** [comment]
I think watermark and further cleanup could be done later in following prs

**Pull request merged**